assembl/static2/js/app/components/debate/common/topPostForm.jsx

```diff
@@ -137,6 +140,9 @@ class TopPostForm extends React.Component<TopPostFormProps, TopPostFormState> {
             this.setState({ submitting: false });
           });
       });
+    } else if (publicationState === PublicationStates.DRAFT && (!subject && bodyIsEmpty)) {
+      displayAlert('warning', I18n.t('debate.brightMirror.fillEitherTitleContent'));
+      this.setState({ submitting: false });
```

Comment:
Functional form, perhaps. https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b

Reply:
Isn't it only useful when you use the previousState (eg to toggle a boolean or increase a counter) ?